hazelcast/src/main/java/com/hazelcast/dataconnection/impl/DataConnectionServiceImpl.java

```diff
@@ -166,7 +166,8 @@ public String typeForDataConnection(String name) {
 
     @Override
     public Class<? extends DataConnection> classForDataConnectionType(String type) {
-        Class<? extends DataConnection> dataConnectionClass = typeToDataConnectionClass.get(type);
+        String caseInsensitiveType = type.toLowerCase(Locale.ROOT);
```

Comment:
we lack to lower in `processDataConnectionRegistrations`: `dataConnectionClassToType.put(registration.clazz(), registration.type());`